transport-native-epoll/src/main/c/netty_epoll_linuxsocket.c

```diff
@@ -481,7 +481,7 @@ jint netty_epoll_linuxsocket_JNI_OnLoad(JNIEnv* env, const char* packagePrefix)
     return NETTY_JNI_VERSION;
 }
 
-void netty_epoll_linuxsocket_JNI_OnUnLoad(JNIEnv* env) {
+void __attribute__((visibility("hidden"))) netty_epoll_linuxsocket_JNI_OnUnLoad(JNIEnv* env) {
```

Comment:
Alternatively it looks like you can make the default visibility "hidden" via a compiler flag `-fvisibility` [1][2]. The gcc wiki even recommends using `hidden` as the default and then explicitly adding `__attribute__ ((visibility ("default")))` when you need to export symbols ... did you consider this approach?

[1] https://gcc.gnu.org/wiki/Visibility
[2] https://clang.llvm.org/docs/LTOVisibility.html

Reply:
I can try this... will see how to do something similar with windows for tcnative later if it works